### PR TITLE
Support JIT for SBT

### PR DESCRIPTION
**Pull request by zitongzhan**

### CODE REVIEW

File: pypose/sparse/sparse_block_tensor.py [hunk 1]
```diff
@@ -1,3 +1,8 @@
+from math import prod
+from typing import Any, List, Optional, Tuple
+import torch
+from torch import Tensor, tensor, jit, zeros, arange, meshgrid, ones, stack, \
```
Comment (huyaoyu) [inline pypose/sparse/sparse_block_tensor.py:4]: Is this required by the JIT feature of PyTorch, or it is required by the contributing guideline?

File: pypose/sparse/sparse_block_tensor.py [hunk 2]
```diff
@@ -103,7 +109,8 @@ def make_coo_indices_and_dims_from_hybrid(hybrid):
 
     return indices_new, coo_dim
 
-def repeated_value_as_hybrid_value(coo, block_shape, val_func):
+@jit.script
+def repeated_value_as_hybrid_value(coo: Tensor, block_shape: List[int], dtype:Optional[torch.dtype]=None, mode:str='zeros'):
```
Comment (huyaoyu) [inline pypose/sparse/sparse_block_tensor.py:113]: Need to modify the doc string to reflect the change in the argument list.

File: pypose/sparse/sparse_block_tensor.py [hunk 3]
```diff
@@ -134,20 +141,27 @@ def repeated_value_as_hybrid_value(coo, block_shape, val_func):
...
+    val = val[..., None, None].tile(block_shape)
 
     # Only use the sparse dimension.
-    s_dim = coo.shape[:2]
+    s_dim: List[int] = list(coo.shape[:2])
 
-    return torch.sparse_coo_tensor(
-            coo.indices(), val, size=(*s_dim, *block_shape)
+    return sparse_coo_tensor(
+            coo.indices(), val, size=s_dim + block_shape
```
Comment (huyaoyu) [inline pypose/sparse/sparse_block_tensor.py:160]: Is `list(coo.shape[:2])` required by JIT. Also, is `s_dim + block_shape` required by JIT?

File: pypose/sparse/sparse_block_tensor.py [hunk 4]
```diff
@@ -134,20 +141,27 @@ def repeated_value_as_hybrid_value(coo, block_shape, val_func):
...
         ).coalesce()
 
-def sparse_coo_2_hybrid_block_sequence(coo, block_shape):
+@jit.script
+def sparse_coo_2_hybrid_block_sequence(coo, block_shape: List[int]):
```
Comment (huyaoyu) [inline pypose/sparse/sparse_block_tensor.py:164]: Need to update the doc string.

**Comment by zitongzhan** [comment]
Docstring is now consistent with API. `sparse_coo_2_hybrid_block_sequence` is removed to simplify the calling logic.

File: pypose/sparse/sparse_block_tensor.py [hunk 5]
```diff
@@ -90,108 +97,74 @@ def make_coo_indices_and_dims_from_hybrid(hybrid):
...
     n_block = coo.values().shape[0]
-    n_block_elem = block_shape[0] * block_shape[1]
 
     # Prepare the sequence number.
-    val = val_func(n_blk=n_block, device=coo.device)
-    val = val.repeat_interleave(n_block_elem).view((n_block, *block_shape))
+    if mode == 'zeros':
+        val = zeros(n_block, device=coo.device, dtype=dtype)
+    elif mode == 'ones':
+        val = ones(n_block, device=coo.device, dtype=dtype)
```
Comment (huyaoyu) [inline pypose/sparse/sparse_block_tensor.py:152]: We actually do now have any use case for the `'ones'` mode.

**Review by huyaoyu — APPROVED** [review]
With minor comments.

**Pull request merged**